.github/workflows/tox.yml

```
@@ -238,3 +238,50 @@ jobs:
         uses: codecov/codecov-action@015f24e6818733317a2da2edd6290ab26238649a  # v5.0.7
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+
+      - if: ${{ (success() || failure()) && contains(matrix.coverage, 'github') && matrix.pytest == 'true' }}
+        name: add run info to coverage filename
+        run: mv $(find ~ -name .coverage) .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
```

I seem to be having trouble with this line.

I am trying this out on the dkist repo: https://github.com/DKISTDC/dkist/pull/471 and by default tox puts `.coverage` in the `.tmp/py312/` directory. Even if I move the `.coverage` file up into the root repo directory this still doesn't seem to find it.

Should this in fact be this:

```suggestion
        run: mv $(find . -name .coverage) .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
```

i.e. look in the current directory?

---

ah, good catch, it should be using the working directory

---

What I can't figure out is if there's an way of doing the equivalent of this:
```
            rf"--cov-report=xml:${{GITHUB_WORKSPACE}}{sep}coverage.xml ")
```

Where we can set the .coverage path?

---

`coverage` makes a `.coverage` database by default, if you don't specify a `--cov-report` type. You CAN specify `--cov-report=` to be `html xml json lcov annotate` (see https://pytest-cov.readthedocs.io/en/latest/reporting.html), but as far as I can tell the markdown table generator can only ingest the `.coverage` database

---

Yeah, it's the inability to specify the path of the coverage db that seems to be a problem.

Even with the last commit I don't seem to be finding the db file: https://github.com/DKISTDC/dkist/actions/runs/12161554758/job/33916444661?pr=471#step:14:1

Testing this out locally *seems* to work.

---

I vaguely remember running into this issue where the working directory (and thus the `.coverage` file) was somewhere unexpected. Now that I think about it, that may be why I used `find ~`; I wanted to recursively find the `.coverage` file wherever it was

---

I hope you don't mind, but I pushed a couple of experimental commits to make it search from the root workspace dir down. Let's see what happens.

---

no problem!

---

Ok well the plot thickens: https://github.com/DKISTDC/dkist/actions/runs/12161554758/job/33918322027#step:14:1

The find command finds the file, but then it fails to upload?

---

I tried to go a different way and let the upload action figure it out: https://github.com/DKISTDC/dkist/actions/runs/12161554758/job/33921111684#step:14:16

It did not go well.